src/databricks/labs/ucx/install.py

```diff
@@ -676,17 +675,39 @@ def _get_workspace(workspace_id: int, ids_to_workspace: dict[int, Workspace]) ->
             msg = "Current workspace is not known, Please run as account-admin: databricks labs ucx sync-workspace-info"
             raise KeyError(msg) from None
 
-    def _get_workspace_info(self, current_workspace_id: int):
+    def _get_workspace_ids(self):
         account_client = self._get_safe_account_client()
-        workspace = account_client.workspaces.get(current_workspace_id)
-        current_workspace_client = account_client.get_workspace_client(workspace)
-        installation = Installation.current(current_workspace_client, self.product_info.product_name())
-        workspace_info = WorkspaceInfo(installation, current_workspace_client)
-        return workspace_info.load_workspace_info()
+        workspaces = account_client.workspaces.list()
+        ids_to_workspace = {}
+        for workspace in workspaces:
+            ids_to_workspace[workspace.workspace_id] = workspace
+        return ids_to_workspace
+
+    def get_workspace_contexts(self, collection_workspace_id: int) -> list[WorkspaceContext]:
```

Comment:
```suggestion
    def get_workspace_contexts(self, other_workspace_id: int) -> list[WorkspaceContext]:
```

i wonder if `other workspace` is a better name here to reason about

Reply:
done